### ISSUE TITLE
Possible XXE attack?

### ISSUE THREAD

**Issue by amihaiemil** [zerocracy/farm#1730]
Certain users can upload XML files and this **might** pose a very high rish, because our XML parser may also process [XML External Entities](https://www.owasp.org/index.php/XML_External_Entity_(XXE)_Processing).
 
I didn't try it since I didn't want to take down Production, but basically if the user uploads this kind of XXE xml, they could either get access to internal Files (from the OS) or take down Prod via OutOfMemory errors (see the ``lolz`` XXE example).

A solution to this is to use formulars instead of letting the user upload raw XMLs.

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) please, pay attention to this issue

**Comment by g4s8** [comment]
@amihaiemil most probably this kind of atack should be prevented by saxon XML parser, but we need to verify that.

**Comment by amihaiemil** [comment]
@g4s8 Yes, we can verify, but I'm not sure the parser prevents it. This is because the loading of "external entities" is a functionality of XML, not necessarily a vulnerability: it becomes a vulnerability only when you expose it to the world and everyone can "load" any external entity they want. On the other hand, if you keep it internal, there shouldn't be any problem :)

**Comment by 0crat** [comment]
Job [#1730](https://github.com/zerocracy/farm/issues/1730) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://www.zerocracy.com/policy.html#29): +15 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)
